lib/widgets/app.dart

```diff
@@ -57,22 +58,51 @@ class ChooseAccountPage extends StatelessWidget {
     assert(!PerAccountStoreWidget.debugExistsOf(context));
     final globalStore = GlobalStoreWidget.of(context);
     return Scaffold(
-      appBar: AppBar(title: const Text('Choose account')),
-      body: Center(
-        child: ConstrainedBox(
-          constraints: const BoxConstraints(maxWidth: 400),
-          child: Column(mainAxisAlignment: MainAxisAlignment.center, children: [
-            for (final (:accountId, :account) in globalStore.accountEntries)
-              _buildAccountItem(context,
-                accountId: accountId,
-                title: Text(account.realmUrl.toString()),
-                subtitle: Text(account.email)),
-            const SizedBox(height: 12),
-            ElevatedButton(
-              onPressed: () => Navigator.push(context,
-                AddAccountPage.buildRoute()),
-              child: const Text('Add an account')),
-          ]))));
+      appBar: AppBar(
+        title: const Text('Choose account'),
+        actions: const [ChooseAccountPageOverflowButton()]),
+      body: SafeArea(
+        minimum: const EdgeInsets.all(8),
+        child: Center(
+          child: ConstrainedBox(
+            constraints: const BoxConstraints(maxWidth: 400),
+            child: Column(mainAxisAlignment: MainAxisAlignment.center, children: [
+              for (final (:accountId, :account) in globalStore.accountEntries)
+                _buildAccountItem(context,
+                  accountId: accountId,
+                  title: Text(account.realmUrl.toString()),
+                  subtitle: Text(account.email)),
+              const SizedBox(height: 12),
+              ElevatedButton(
+                onPressed: () => Navigator.push(context,
+                  AddAccountPage.buildRoute()),
+                child: const Text('Add an account')),
+            ]))),
+      ));
+  }
+}
+
+enum ChooseAccountPageOverflowMenuItem { aboutZulip }
+
+class ChooseAccountPageOverflowButton extends StatelessWidget {
+  const ChooseAccountPageOverflowButton({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    final navigator = Navigator.of(context);
```

Comment:
Interesting! In the common case of an event handler that just has to walk the tree once (it doesn't need the `NavigatorState` for two or more tasks), it might make sense to favor `Navigator.push(context, …)` over `Navigator.of(context).push(…)`, which would prevent the tree-walking work from straying outside the event handler, as it did in this revision.